controllers/reaper_controller_test.go

```diff
@@ -149,35 +149,60 @@ var _ = Describe("Reaper controller", func() {
 
 		By("check that the deployment is created")
 		deploymentKey := types.NamespacedName{Namespace: ReaperNamespace, Name: ReaperName}
-		deployment := &appsv1.Deployment{}
+		currentDeployment := &appsv1.Deployment{}
 
 		Eventually(func() error {
-			return k8sClient.Get(context.Background(), deploymentKey, deployment)
+			return k8sClient.Get(context.Background(), deploymentKey, currentDeployment)
 		}, timeout, interval).Should(Succeed(), "deployment creation check failed")
 
-		Expect(len(deployment.OwnerReferences)).Should(Equal(1), "deployment owner reference not set")
-		Expect(deployment.OwnerReferences[0].UID).Should(Equal(reaper.GetUID()), "deployment owner reference has wrong uid")
+		Expect(len(currentDeployment.OwnerReferences)).Should(Equal(1), "deployment owner reference not set")
+		Expect(currentDeployment.OwnerReferences[0].UID).Should(Equal(reaper.GetUID()), "deployment owner reference has wrong uid")
 
 		By("update deployment to be ready")
-		patchDeploymentStatus(deployment, 1, 1)
+		patchDeploymentStatus(currentDeployment, 1, 1)
 
 		verifyReaperReady(types.NamespacedName{Namespace: ReaperNamespace, Name: ReaperName})
 
+		// Simulate changing the deployment strategy to rolling.
+		// The deployment strategy should be reconciled back to recreate.
+		By("patch deployment to have a rolling update strategy")
+		Eventually(func() error {
+			return k8sClient.Get(context.Background(), deploymentKey, currentDeployment)
+		}, timeout, interval).Should(Succeed(), "could not obtain current deployment")
+		deploymentPatch := client.MergeFrom(currentDeployment.DeepCopy())
```

Comment:
Here's a question.. this patching method here uses different MergeFrom than the one in the actual implementation. Was there a reason for that?

Reply:
This is the original patching method that was used throughout as it was all that the previous controller-runtime version supported. I haven't changed it.

So (to answer a question with a question, sorry): why did you think we should change to doing a strategic merge patch in the first place? I didn't fully understand this as I'm not sure what the difference between MergeFrom and StrategicMergeFrom is. Do you have any ideas on that?